frontend/src/components/Form/SurveyForm.tsx

```diff
@@ -0,0 +1,187 @@
+    if (!formData.improvements.trim()) errors.improvements = 'To pole jest wymagane.'
+
+    return errors
+  }
+
+  const handleSubmitForm = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault()
+
+    const newErrors = validateFormData(formData)
+
+    setFormErrors(newErrors)
+
+    const hasErrors = Object.values(newErrors).some((msg) => msg !== '')
+    if (hasErrors) return
+
+    setFormIsSubmitting(true)
+
+    try {
+      await submitForm(formData)
+    } catch (error) {
+      if (error instanceof Error) {
+        console.error(error.message)
+      } else {
+        console.error('An unknown error occurred')
+      }
+    } finally {
+      setFormIsSubmitting(false)
+    }
+  }
+
+  const handleCheckboxChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, checked } = event.target
+    setFormData((prev) => ({
+      ...prev,
+      tasks: {
+        ...prev.tasks,
+        [name]: checked,
+      },
+    }))
+  }
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+    const { name, value } = event.target
+    setFormData((prev) => ({
+      ...prev,
+      [name]: value,
+    }))
+  }
+
+  return (
+    <div>
+      <Typography variant="h1">Formul.ai</Typography>
+      <Stack component="form" onSubmit={handleSubmitForm} spacing={1}>
+        <FormLabel error={!!formErrors.tasks}>Co masz robić w projekcie</FormLabel>
+        <FormGroup>
+          {Object.keys(formData.tasks).map((task) => (
+            <FormControlLabel
+              key={task}
+              control={
+                <Checkbox
+                  name={task}
+                  checked={formData.tasks[task as keyof typeof formData.tasks]}
+                  onChange={handleCheckboxChange}
+                />
+              }
+              label={task}
+            />
+          ))}
+        </FormGroup>
+        {formErrors.tasks && (
+          <Typography color="error" variant="caption">
+            {formErrors.tasks}
+          </Typography>
+        )}
+
+        <FormLabel error={!!formErrors.rating}>Jak oceniasz proces do tego momentu</FormLabel>
+        <Grid container justifyContent="center">
+          <RadioGroup row name="rating" value={formData.rating} onChange={handleChange}>
+            {Array.from({ length: 10 }, (_, i) => (
+              <FormControlLabel
+                key={i}
+                value={String(i + 1)}
+                control={<Radio sx={{ p: 0.5 }} />}
+                label={i + 1}
+                labelPlacement="bottom"
+                sx={{ mx: 0.5 }}
+              />
+            ))}
+          </RadioGroup>
+        </Grid>
+        {formErrors.rating && (
+          <Typography color="error" variant="caption">
+            {formErrors.rating}
+          </Typography>
+        )}
+
+        <FormLabel error={!!formErrors.likes}>Co Ci się podoba</FormLabel>
```

Comment:
TextField has label prop supported - also [covers red when invalid](https://mui.com/material-ui/react-text-field/#validation)


```suggestion
          <TextField
          label="Co Ci się podoba"
          multiline
          rows={4}
          name="likes"
          value={formData.likes}
          onChange={handleChange}
          error={!!formErrors.likes}
          helperText={formErrors.likes}
        />

```